vmm/src/console_devices.rs

```diff
@@ -235,9 +235,32 @@ pub fn pre_create_console_devices(vmm: &mut Vmm) -> ConsoleDeviceResult<ConsoleI
             vmconfig.serial.file = Some(path.clone());
         }
         ConsoleOutputMode::Tty => {
-            let out = stdout();
-            console_info.serial_main_fd = Some(out.as_raw_fd());
-            set_raw_mode(&out, vmm.original_termios_opt.clone())?;
+            // Duplicating the file descriptors like this is needed as otherwise
+            // they will be closed on a reboot and the numbers reused
+
+            // SAFETY: FFI call to dup. Trivially safe.
+            let stdout = unsafe { libc::dup(libc::STDOUT_FILENO) };
```

Comment:
You can use something like `stdout().as_raw_fd()` to get the fd number.

Can you go into more details as to why closing the fd will be a problem?